packages/api-clients/open-api/authorizationApi.yml

```
@@ -2191,9 +2205,25 @@ components:
       required:
         - purposeId
     ProducerKeychain:
+      description: |
+        Models a producer keychain that can be either a full producer keychain or a partial producer keychain based on the caller
+        tenant. In case the caller tenant is the producer of the keychain (i.e., the owner), the
+        full keychain is returned, otherwise a partial keychain is returned.
+      oneOf:
+        - $ref: "#/components/schemas/FullProducerKeychain"
+        - $ref: "#/components/schemas/PartialProducerKeychain"
+      discriminator:
+        propertyName: visibility
+        mapping:
+          PARTIAL: "#/components/schemas/PartialProducerKeychain"
+          FULL: "#/components/schemas/FullProducerKeychain"
+    FullProducerKeychain:
       description: Models Producer Keychain details
       type: object
```

As done in FullClient, let's add a description

```suggestion
      type: object
     description: |
        Models a producer keychain.
        The owner of the producer keychain is the producer tenant, identified by the producerId.
        The only fields that are visible to any caller tenant are the ones that are
        present also in the PartialProducerKeychain schema. The rest are visible only to the owner tenant.
```